### PR TITLE
Support for pointers and refs for all types

### PR DESCRIPTION
**Pull request by varunagrawal**
Both `_QualifiedType` and `_BasisType` now supports the raw pointer, shared pointer and reference syntax.

I tried abstracting it all out to `Type` but the parser fails, so will tackle that in another PR.

### CODE REVIEW

**Review by ProfFan — APPROVED** [review]
I think the MATLAB unit tests has not been updated yet, but the syntax looks good to me :)

File: tests/expected-matlab/+gtsam/Point2.m [hunk 1]
```diff
@@ -7,6 +7,12 @@
 %
 %-------Methods-------
 %argChar(char a) : returns void
+%argChar(char a) : returns void
+%argChar(char a) : returns void
```
Comment (ProfFan) [inline tests/expected-matlab/+gtsam/Point2.m:11]: Overload resolution in MATLAB wrapper is still an issue (my bad).

**Comment by varunagrawal** [comment]
Should I merge this?

**Comment by ProfFan** [comment]
Do you want to fix the MATLAB overload resolution in a separate PR (or ignore it)?

**Comment by varunagrawal** [comment]
I'll save it for later. I am too tired :(

**Pull request merged**